Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -79,16 +79,22 @@ public final class MapboxMap: MapboxMapProtocol {
     // MARK: - Style loading
 
     private func observeStyleLoad(_ completion: @escaping (Result<Style, Error>) -> Void) {
-        onNext(event: .styleLoaded) { [style] _ in
+        let cancellable = CompositeCancelable()
+
+        cancellable.add(onNext(event: .styleLoaded) { [style] _ in
             if !style.isLoaded {
                 Log.warning(forMessage: "style.isLoaded == false, was this an empty style?", category: "Style")
             }
             completion(.success(style))
-        }
+            cancellable.cancel()
+        })
+
+        cancellable.add(onEvery(event: .mapLoadingError) { event in
```

Comment:
Why did we change `onNext` to `onEvery`?

Reply:
there might be other error represented by `mapLoadingError` arrive before style load error arrives, so I want to make sure we are catching error event indicating the given style cannot be loaded here.

Reply:
Makes sense, good catch 👍